### PR TITLE
[18.0][MIG] agreement: Migration to 18.0

### PR DESCRIPTION
**Pull request by LucasTran380381**

### CODE REVIEW

**Review by rousseldenis — APPROVED** [review]
LGTM

**Review by ajaniszewska-dev — APPROVED** [review]
LG

File: agreement/models/agreement.py [hunk 1]
```diff
@@ -51,8 +51,8 @@ class Agreement(models.Model):
     @api.model
     def _domain_selection(self):
         return [
-            ("sale", _("Sale")),
-            ("purchase", _("Purchase")),
+            ("sale", self.env._("Sale")),
```
Comment (sebalix) [inline agreement/models/agreement.py:53]: Is it `pre-commit` updating such statements? Or should it be part of the migration commit?

File: agreement/views/agreement.xml.orig [hunk 2]
```diff
@@ -0,0 +1,135 @@
+<?xml version="1.0" encoding="utf-8" ?>
```
Comment (sebalix) [inline agreement/views/agreement.xml.orig]: This file should be removed (local copy?)

**Review by sebalix — COMMENTED** [review]
LG, pre-commit commit to clean up (there was a file added to it, then removed in migration commit) + the translation function calls (but not blocking on this).

File: agreement/models/agreement.py (hunk 1 above)
Comment (LucasTran380381) [reply agreement/models/agreement.py:53]: this should be part of migration commit. I will fix it

File: agreement/views/agreement.xml.orig (hunk 2 above)
Comment (LucasTran380381) [reply agreement/views/agreement.xml.orig]: I will removed it

**Review by sebalix — APPROVED** [review]
Thank you @LucasTran380381 , LG

**Pull request merged**